nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-restapi-processors/src/test/java/org/apache/nifi/processors/elasticsearch/PutElasticsearchRecordTest.java

```diff
@@ -0,0 +1,850 @@
+        runner.assertTransferCount(PutElasticsearchRecord.REL_SUCCESSFUL_RECORDS, 0);
+        runner.assertTransferCount(PutElasticsearchRecord.REL_ERROR_RESPONSES, 0);
+    }
+
+    @Test
+    public void testIncompatibleTimestampRecordPath() throws Exception {
+        clientService.setEvalConsumer((List<IndexOperationRequest> items) -> {
+            long timestampCount = items.stream().filter(item ->  "Hello".equals(item.getFields().get("@timestamp"))).count();
+            assertEquals(1, timestampCount);
+        });
+
+        registry.addSchema(RECORD_PATH_TEST_SCHEMA, recordPathTestSchema);
+        runner.setProperty(PutElasticsearchRecord.AT_TIMESTAMP_RECORD_PATH, "/msg");
+        runner.enqueue(Paths.get(TEST_DIR, "5_flowFileContents.json"), Collections.singletonMap("schema.name", RECORD_PATH_TEST_SCHEMA));
+        runner.run();
+        runner.assertTransferCount(PutElasticsearchRecord.REL_SUCCESS, 1);
+        runner.assertTransferCount(PutElasticsearchRecord.REL_FAILURE, 0);
+        runner.assertTransferCount(PutElasticsearchRecord.REL_RETRY, 0);
+        runner.assertTransferCount(PutElasticsearchRecord.REL_FAILED_RECORDS, 0);
+        runner.assertTransferCount(PutElasticsearchRecord.REL_SUCCESSFUL_RECORDS, 0);
+        runner.assertTransferCount(PutElasticsearchRecord.REL_ERROR_RESPONSES, 0);
+    }
+
+    @Test
+    public void testNotFoundELRecordPaths() throws Exception {
+        clientService.setEvalConsumer((List<IndexOperationRequest> items) -> {
+            long nullIdCount = items.stream().filter(item ->  item.getId() == null).count();
+            long noTimestampCount = items.stream().filter(item -> !item.getFields().containsKey("@timestamp")).count();
+            assertEquals(1, nullIdCount, getUnexpectedCountMsg("null id"));
+            assertEquals(1, noTimestampCount, getUnexpectedCountMsg("noTimestamp"));
+        });
+
+        registry.addSchema(RECORD_PATH_TEST_SCHEMA, recordPathTestSchema);
+        runner.removeProperty(PutElasticsearchRecord.AT_TIMESTAMP);
+        runner.setProperty(PutElasticsearchRecord.ID_RECORD_PATH, "${id_not_exist}");
+        runner.setProperty(PutElasticsearchRecord.AT_TIMESTAMP_RECORD_PATH, "${not_exist}");
+        runner.enqueue(Paths.get(TEST_DIR, "6_flowFileContents.json"), Collections.singletonMap("schema.name", RECORD_PATH_TEST_SCHEMA));
+        runner.run();
+        runner.assertTransferCount(PutElasticsearchRecord.REL_SUCCESS, 1);
+        runner.assertTransferCount(PutElasticsearchRecord.REL_FAILURE, 0);
+        runner.assertTransferCount(PutElasticsearchRecord.REL_RETRY, 0);
+        runner.assertTransferCount(PutElasticsearchRecord.REL_FAILED_RECORDS, 0);
+        runner.assertTransferCount(PutElasticsearchRecord.REL_SUCCESSFUL_RECORDS, 0);
+        runner.assertTransferCount(PutElasticsearchRecord.REL_ERROR_RESPONSES, 0);
+    }
+
+    @Test
+    public void testEmptyELRecordPaths() throws Exception {
```

Comment:
Lot of duplicated code with `testNotFoundELRecordPaths`, should we combine these into a common `testELRecordPaths` that each of these test cases calls, setting any specfic properties, etc.?

Reply:
Done